File: packages/execution-engine/src/models/risk_validation_result.py
from enum import Enum
from typing import List, Optional

from pydantic import BaseModel


class RiskRuleName(str, Enum):
    MAX_RISK_PER_TRADE = "max_risk_per_trade"
    DAILY_LOSS_LIMIT = "daily_loss_limit"
    MAX_POSITIONS = "max_positions"
    MAX_LOT_EXPOSURE = "max_lot_exposure"
    SPREAD_CHECK = "spread_check"
    SLIPPAGE_CHECK = "slippage_check"
    TRAILING_DRAWDOWN = "trailing_drawdown"


class RiskRuleResult(BaseModel):
    rule: RiskRuleName
    passed: bool
    reason: Optional[str] = None


class RiskValidationResult(BaseModel):
    approved: bool
    signal_id: str
    account_id: str
    rules_checked: List[RiskRuleResult]
    rejected_by: Optional[RiskRuleName] = None
    validated_at: str  # ISO 8601

File: packages/backend/src/common/types/risk-validation-result.ts
export type RiskRuleName =
  | 'max_risk_per_trade'
  | 'daily_loss_limit'
  | 'max_positions'
  | 'max_lot_exposure'
  | 'spread_check'
  | 'slippage_check';

export interface RiskRuleResult {
  rule: RiskRuleName;
  passed: boolean;
  reason?: string;
}

export interface RiskValidationResult {
  approved: boolean;
  signal_id: string;
  account_id: string;
  rules_checked: RiskRuleResult[];
  rejected_by?: RiskRuleName;
  validated_at: string;          // ISO 8601
}
